### PR TITLE
Changing interface from worker.channels.on to worker.on

### LINKED ISSUE #10: cworker.channels.on and cworker.on will lead to bugs
I suspect developers will forget to do `cworker.channels.on` and instead do `cworker.on`. We've done this ourselves. We should explore a fix.

The main motivation for adding the channels vs. just using on is to avoid introducing vulnerabilities where the child worker emits `running` or other kinds of internal events. Maybe if `cworker.on` is used on none of the standard channels we should throw an exception? @annieliu @olegpesok Thoughts?

### PR DESCRIPTION
**Pull request by olegpesok**
One caveat here is that we've remapped the internal signal 'error' to
the worker channel 'error'. This is non-obvious behavior, needs more
docs. Closes #10. Depends on https://github.com/GitStarInc/espectro-worker/pull/12

### CODE REVIEW

**Comment by olegpesok** [comment]
@deian @annieliu take a look at new iface.

File: index.js [hunk 1]
```diff
@@ -12,29 +13,48 @@ function init(self) {
...
+        debug('WARNING: \'error\' channel is reserved, ' +
+            'mapping to internal error listener.');
+        return f.call(self._espectro, ch, cb);
+      } else {
+        return f.call(self, ch, cb);
+      }
+    };
+  }
+
+  self.on = self.addListener = errorWrap(self.addListener);
```
Comment (deian) [inline index.js]: - move `errorWrap` to global scope
- After `util.inhertis(...)`, modify the `on` on the prototype. That is `CWorker.prototype.on = function (...) {` should do the wrapping.

File: prelude.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ var ro       = {};
       return _rpc[name];
     }
     _rpc[name] = function () {
-      let args = Array.prototype.splice.call(arguments, 0);
```
Comment (deian) [inline prelude.js]: bah, how did we miss this again? We should add a test for this.

File: README.md [hunk 3]
```diff
@@ -7,18 +7,18 @@ Library used to create named channels and RPC's between a worker and it's parent
 
 ### Asynchronous channels
 
-Worker channels are `EventEmitters` which you can use to communicate with the worker.  The interface is mostly standard (see [node](http://nodejs.org/api/events.html)'s interface).  It differs only in that `emit` sends the message to the worker (correspondingly, parent) instead of invoking a local callback `on`.
+Worker channels are `EventEmitters` which you can use to communicate with the worker.  The interface is mostly standard (see [node](http://nodejs.org/api/events.html)'s interface).  It differs only in that `emit` sends the message to the worker (correspondingly, parent) instead of invoking a local callback `on`. *NOTE* the 'error' channel is reserved and is automatically remapped to the internal 'error' listener.
```
Comment (deian) [inline README.md]: We should update readme to describe the internal `_espectro` stuff.

File: index.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ const EventEmitter = require('events').EventEmitter;
 let   CWorker      = require('espectro-worker');
 const ro           = require('./lib/remote-object.js');
 const callback     = require('./lib/callback.js');
+const debug        = require('debug')('espectro-channels');
```
Comment (deian) [inline index.js]: Add to `package.json`; fails tests cause of this.

File: prelude.js (hunk 2 above)
Comment (olegpesok) [inline prelude.js]: It's pretty much equivalent functionality so it's hard to catch. One is just more 'technically' correct as arguments object isn't being modified upstream.

File: README.md (hunk 3 above)
Comment (olegpesok) [inline README.md]: That should go into espectro-worker README...or here as well?

Comment (deian) [inline README.md]: here is where we define some of the events (e.g., 'running') no?

Comment (olegpesok) [inline README.md]: 'running' etc is defined in espectro-worker.

Comment (deian) [inline README.md]: Woops, sorry. In that case no. Maybe just reference espectro-worker then

**Comment by deian** [comment]
This addresses my concerns.

**Pull request closed without merging**